### PR TITLE
feat(commands): add `leon kill`

### LINKED ISSUE #172: [Feature] Command `leon kill`
## Description

We should be able to "kill" (remove) a Leon instance.

## Describe the solution you'd like

With the new command `leon kill`, we will be able to kill a Leon instance.

### Options

- `--name`: specify the Leon instance name to kill

## Describe alternatives you've considered

Manually removing the folder `.leon`, and editing `config/config.json` to remove the entry for the instance.

### PR DESCRIPTION
**Pull request by Divlo**
## What changes this PR introduce?

Add the new command `leon kill`.

## List any relevant issue numbers

Fixes #172 

## Is there anything you'd like reviewers to focus on?

There are no strong automated tests for this feature.
We might want to add some, but to do that, we should find out how to mock the `config`.

### CODE REVIEW

File: src/commands/kill.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  async execute(): Promise<number> {
+    try {
+      const leonInstance = await LeonInstance.get(this.name)
+      await leonInstance.kill()
+      console.log(`Leon instance "${leonInstance.name}" killed.`)
+      return 0
+    } catch (error) {
+      await log.error({
+        error,
+        commandPath: 'start'
```
Comment (ArthurDelamare) [inline src/commands/kill.ts:26]: Don't you mean ```commandPath: 'kill'``` here ?

Comment (Divlo) [reply src/commands/kill.ts]: Ah, yep good catch, thanks. :smile:
Updated.

**Review by ArthurDelamare — COMMENTED** [review]
The logic seems fine to me 😄 

```LeonInstance.get(this.name)``` is checking if the instance exist, I guess we don't need to make another check in the kill method (unless the config file is corrupted but it seems extremely unlikely).

**Comment by Divlo** [comment]
> The logic seems fine to me smile
> 
> `LeonInstance.get(this.name)` is checking if the instance exist, I guess we don't need to make another check in the kill method (unless the config file is corrupted but it seems extremely unlikely).

Indeed, it would be better to avoid this thing in `LeonInstance.kill`:
```ts
const instanceIndex = instances.findIndex((instance) => {
  return instance.name === this.name
})
```
but we need to get the `instanceIndex` to remove it in the config.
At least it does an extra check if the config is corrupted, even if it is extremely unlikely.

Do you have another solution? Or is it fine to merge this feature? :smile:

**Review by ArthurDelamare — APPROVED** [review]

**Pull request merged**